sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/ContainerResource.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.resources;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/** Factory for {@link Resource} retrieving Container ID information. */
+public final class ContainerResource {
+
+  private static final String UNIQUE_HOST_NAME_FILE_NAME = "/proc/self/cgroup";
+  private static final Pattern HEX_EXTRACTOR =
+      Pattern.compile("^([\\w]*?-)?([a-fA-F0-9]+)(\\.[\\w]*?)?$");
+
+  private final String cgroupFilePath;
+
+  // package private for testing purposes
+  ContainerResource(String cgroupFilePath) {
+    this.cgroupFilePath = cgroupFilePath;
+  }
+
+  /** Returns resource with container information. */
+  public static Resource get() {
+    ContainerResource factory = new ContainerResource(UNIQUE_HOST_NAME_FILE_NAME);
+    String containerId = factory.extractContainerId();
+
+    if (containerId == null) {
+      return Resource.empty();
+    }
+    return Resource.create(
+        Attributes.builder().put(ResourceAttributes.CONTAINER_ID, containerId).build());
```

Comment:
this can be simplified to just `Attributes.of(...)`

Reply:
done